src/Modules/OrchardCore.Commerce.Payment.Stripe/Services/StripePaymentService.cs

```diff
@@ -215,9 +216,51 @@ public async Task<PaymentIntent> CreatePaymentIntentAsync(Amount total)
         return paymentIntent;
     }
 
+    public async Task<PaymentIntentConfirmOptions> GetStripeConfirmParametersAsync(string middlewareUrl)
+    {
+        var order = await _contentManager.NewAsync(Commerce.Abstractions.Constants.ContentTypes.Order);
+        await _paymentService.UpdateOrderWithDriversAsync(order);
+
+        var part = order.As<OrderPart>();
+        var billing = part.BillingAddress.Address ?? new Address();
+        var shipping = part.ShippingAddress.Address ?? new Address();
+
+        var model = new PaymentIntentConfirmOptions
+        {
+            ReturnUrl = middlewareUrl,
+            PaymentMethodData = new PaymentIntentPaymentMethodDataOptions
+            {
+                BillingDetails = new PaymentIntentPaymentMethodDataBillingDetailsOptions
+                {
+                    Email = part.Email?.Text,
+                    Name = billing.Name,
+                    Phone = part.Phone?.Text,
+                    Address = CreateAddressOptions(billing),
+                },
+            },
+            Shipping = new ChargeShippingOptions
+            {
+                Name = shipping.Name,
+                Phone = part.Phone?.Text,
+                Address = CreateAddressOptions(shipping),
+            },
+        };
+        return model;
+    }
+
+    private static AddressOptions CreateAddressOptions(Address address) =>
+        new()
+        {
+            City = address.City ?? string.Empty,
+            Country = address.Region ?? string.Empty,
+            Line1 = address.StreetAddress1 ?? string.Empty,
+            Line2 = address.StreetAddress2 ?? string.Empty,
+            PostalCode = address.PostalCode ?? string.Empty,
+            State = address.Province ?? string.Empty,
+        };
     private async Task<PaymentIntent> GetOrUpdatePaymentIntentAsync(
-        string paymentIntentId,
-        Amount defaultTotal)
+    string paymentIntentId,
+    Amount defaultTotal)
```

Comment:
Revert this change, parameters should be indented from the starting line of the function signature.